tests/destinations/test_path_utils.py

```diff
@@ -51,19 +127,168 @@ def test_get_table_prefix_layout() -> None:
+        current_datetime=now,
+        load_package_timestamp=now_timestamp,
+        file_name=job_info.file_name(),
+    )
+
+    # expect only one call
+    assert calls == 1
+
+    # If the value for current_datetime is not pendulum.DateTime
+    # it should fail with RuntimeError exception
+    with pytest.raises(AttributeError):
+        create_path(
+            "{schema_name}/{table_name}/{load_id}.{file_id}.{timestamp}.{ext}",
+            schema_name="schema_name",
+            load_id=load_id,
+            current_datetime="now",  # type: ignore
+            load_package_timestamp=now_timestamp,
+            file_name=job_info.file_name(),
+        )
+    with pytest.raises(RuntimeError):
+        create_path(
+            "{schema_name}/{table_name}/{load_id}.{file_id}.{timestamp}.{ext}",
+            schema_name="schema_name",
+            load_id=load_id,
+            current_datetime=lambda: 1234,  # type: ignore
+            load_package_timestamp=now_timestamp,
+            file_name=job_info.file_name(),
+        )
+
+
+def test_create_path_uses_current_moment_if_current_datetime_is_not_given(
+    test_load: TestLoad, mocker
+) -> None:
+    load_id, job_info = test_load
+    logger_spy = mocker.spy(logger, "info")
+    pendulum_spy = mocker.spy(pendulum, "now")
+    create_path(
+        "{schema_name}/{table_name}/{load_id}.{file_id}.{timestamp}.{ext}",
+        schema_name="schema_name",
+        load_id=load_id,
+        file_name=job_info.file_name(),
+    )
+    logger_spy.assert_called_once_with("current_datetime is not set, using pendulum.now()")
+    pendulum_spy.assert_called()
+
+
+def test_create_path_uses_load_package_timestamp_as_current_datetime(
+    test_load: TestLoad, mocker
+) -> None:
+    load_id, job_info = test_load
+    now = pendulum.now()
+    now_timestamp = now.to_iso8601_string()
+    logger_spy = mocker.spy(logger, "info")
+    pendulum_spy = mocker.spy(pendulum, "parse")
+    create_path(
+        "{schema_name}/{table_name}/{load_id}.{file_id}.{timestamp}.{ext}",
+        schema_name="schema_name",
+        load_id=load_id,
+        load_package_timestamp=now_timestamp,
+        file_name=job_info.file_name(),
+    )
+    logger_spy.assert_called_once_with(
+        "current_datetime is not set, using timestamp from load package"
+    )
+    pendulum_spy.assert_called_once_with(now_timestamp)
+
+
+def test_create_path_resolves_extra_placeholders(test_load: TestLoad) -> None:
+    load_id, job_info = test_load
+
+    class Counter:
+        def __init__(self) -> None:
+            self.count = 0
+
+        def inc(self, *args, **kwargs):
+            self.count += 1
+
+    counter = Counter()
+    extra_placeholders = {
+        "value": 1,
+        "callable_1": counter.inc,
+        "otter": counter.inc,
+        "otters": "lab",
+        "dlt": "labs",
+        "dlthub": "platform",
+        "x": "files",
+    }
+    now_timestamp = pendulum.now().to_iso8601_string()
+    create_path(
```

Comment:
again, this is not enough. compare to a correct path. and take it by running test, printing out checking manually and then hardcoding. I have not idea if that runs OK or not

Reply:
Done.